File: ALittle/Module/Emulator/Script/IDETreeBool.lua
-- ALittle Generate Lua
module("Emulator", package.seeall)

local ___rawset = rawset
local ___pairs = pairs
local ___ipairs = ipairs


assert(Emulator.IDETreeLogic, " extends class:Emulator.IDETreeLogic is nil")
IDETreeBool = Lua.Class(Emulator.IDETreeLogic, "Emulator.IDETreeBool")

function IDETreeBool:Ctor(ctrl_sys, field_name, rflct, msg, field)
	___rawset(self, "_rflct", rflct)
	___rawset(self, "_msg", msg)
	___rawset(self, "_field", field)
	local value = protobuf.reflection_getbool(self._rflct, self._msg, self._field)
	___rawset(self, "_item", ctrl_sys:CreateControl("ide_common_tree_bool", self))
	self:AddChild(self._item)
	self.width = self._item.width
	self.height = self._item.height
	self._item_title.text = "[" .. protobuf.fielddescriptor_cpptypename(field) .. "] " .. field_name .. ": "
	if value then
		self._value_dropdown.text = "true"
	else
		self._value_dropdown.text = "false"
	end
	self._value_dropdown.width = self.width - self._item_title.width - self._item_title.x - 2
	self._value_dropdown.x = self._item_title.width + self._item_title.x
end

function IDETreeBool:HandleSelectChanegd(event)
	if self._value_dropdown.text == "true" then
		protobuf.reflection_setbool(self._rflct, self._msg, self._field, true)
	else
		protobuf.reflection_setbool(self._rflct, self._msg, self._field, false)
	end
end

File: ALittle/Module/Emulator/JSScript/IDETree/IDETreeBool.js
{
if (typeof Emulator === "undefined") window.Emulator = {};


if (Emulator.IDETreeLogic === undefined) throw new Error(" extends class:Emulator.IDETreeLogic is undefined");
Emulator.IDETreeBool = JavaScript.Class(Emulator.IDETreeLogic, {
	Ctor : function(ctrl_sys, root, field_name, rflct, msg, field) {
		this._rflct = rflct;
		this._msg = msg;
		this._field = field;
		let value = lua.protobuf.reflection_getbool(this._rflct, this._msg, this._field);
		this._item = ctrl_sys.CreateControl("ide_common_tree_bool", this);
		this.AddChild(this._item);
		this.width = this._item.width;
		this.height = this._item.height;
		this._item_title.text = "[" + lua.protobuf.fielddescriptor_cpptypename(field) + "] " + field_name + ": ";
		if (value) {
			this._value_dropdown.text = "true";
		} else {
			this._value_dropdown.text = "false";
		}
		this._value_dropdown.width = this.width - this._item_title.width - this._item_title.x - 2;
		this._value_dropdown.x = this._item_title.width + this._item_title.x;
		this.Init();
		this._value_dropdown.disabled = root.for_show;
	},
	HandleSelectChanegd : function(event) {
		if (this._value_dropdown.text === "true") {
			lua.protobuf.reflection_setbool(this._rflct, this._msg, this._field, true);
		} else {
			lua.protobuf.reflection_setbool(this._rflct, this._msg, this._field, false);
		}
		this.Save();
	},
}, "Emulator.IDETreeBool");

}